grafana-plugin/src/models/label/label.ts

```diff
@@ -26,9 +27,12 @@ export class LabelStore extends BaseStore {
   }
 
   @action.bound
-  async loadValuesForKey(key: ApiSchemas['LabelKey']['id'], search = '') {
+  async loadValuesForKey(
+    key: ApiSchemas['LabelKey']['id'],
+    search = ''
+  ): Promise<components['schemas']['LabelOption']> {
     if (!key) {
-      return [];
+      return {} as components['schemas']['LabelOption'];
```

Comment:
Why we need to cast it here? I can see that after removing casting TS throws the errors but I think for a valid reason. Can we adjust the return type of this function and adapt the places where it's used?

Reply:
I just added the types, didn't really touch this part. It seems it accepts both the given interface and empty values, I assume this is to prevent NPE errors. The cast is really harmless

Reply:
As discussed I believe we should get rid of this condition if it's not necessary.

Everything else looks good 👍 approving PR